data-model/src/main/java/io/micronaut/data/model/schema/DbType.java

```diff
@@ -0,0 +1,340 @@
+
+    /**
+     * A type code representing the generic SQL type {@code JAVA_OBJECT}.
+     *
+     * @see Types#JAVA_OBJECT
+     */
+    JAVA_OBJECT(Types.JAVA_OBJECT),
+
+    /**
+     * A type code representing the generic SQL type {@code DISTINCT}.
+     *
+     * @see Types#DISTINCT
+     */
+    DISTINCT(Types.DISTINCT),
+
+    /**
+     * A type code representing the generic SQL type {@code STRUCT}.
+     *
+     * @see Types#STRUCT
+     */
+    STRUCT(Types.STRUCT),
+
+    /**
+     * A type code representing the generic SQL type {@code ARRAY}.
+     *
+     * @see Types#ARRAY
+     */
+    ARRAY(Types.ARRAY),
+
+    /**
+     * A type code representing the generic SQL type {@code BLOB}.
+     *
+     * @see Types#BLOB
+     */
+    BLOB(Types.BLOB),
+
+    /**
+     * A type code representing the generic SQL type {@code CLOB}.
+     *
+     * @see Types#CLOB
+     */
+    CLOB(Types.CLOB),
+
+    /**
+     * A type code representing the generic SQL type {@code REF}.
+     *
+     * @see Types#REF
+     */
+    REF(Types.REF),
+
+    /**
+     * A type code representing the generic SQL type {@code DATALINK}.
+     *
+     * @see Types#DATALINK
+     */
+    DATALINK(Types.DATALINK),
+
+    /**
+     * A type code representing the generic SQL type {@code BOOLEAN}.
+     *
+     * @see Types#BOOLEAN
+     */
+    BOOLEAN(Types.BOOLEAN),
+
+    /**
+     * A type code representing the generic SQL type {@code ROWID}.
+     *
+     * @see Types#ROWID
+     */
+    ROWID(Types.ROWID),
+
+    /**
+     * A type code representing the generic SQL type {@code NCHAR}.
+     *
+     * @see Types#NCHAR
+     */
+    NCHAR(Types.NCHAR),
+
+    /**
+     * A type code representing the generic SQL type {@code NVARCHAR}.
+     *
+     * @see Types#NVARCHAR
+     */
+    NVARCHAR(Types.NVARCHAR),
+
+    /**
+     * A type code representing the generic SQL type {@code LONGNVARCHAR}.
+     */
+    LONGNVARCHAR(Types.LONGNVARCHAR),
+
+    /**
+     * A type code representing the generic SQL type {@code NCLOB}.
+     *
+     * @see Types#NCLOB
+     */
+    NCLOB(Types.NCLOB),
+
+    /**
+     * A type code representing the generic SQL type {@code XML}.
+     *
+     * @see Types#SQLXML
+     */
+    SQLXML(Types.SQLXML),
+
+    /**
+     * A type code representing the generic SQL type {@code REF CURSOR}.
+     *
+     * @see Types#REF_CURSOR
+     */
+    REF_CURSOR(Types.REF_CURSOR),
+
+    /**
+     * A type code representing identifies the generic SQL type
+     * {@code TIME WITH TIMEZONE}.
+     *
+     * @see Types#TIME_WITH_TIMEZONE
+     */
+    TIME_WITH_TIMEZONE(Types.TIME_WITH_TIMEZONE),
+
+    /**
+     * A type code representing the generic SQL type
+     * {@code TIMESTAMP WITH TIMEZONE}.
+     *
+     * @see Types#TIMESTAMP_WITH_TIMEZONE
+     */
+    TIMESTAMP_WITH_TIMEZONE(Types.TIMESTAMP_WITH_TIMEZONE),
+
+    // Misc types
+
+    /**
+     * A type code representing the generic SQL type {@code UUID}.
+     */
+    UUID(10001),
```

Comment:
What are those constants?

Reply:
I thought we might need some db type to represent UUID, JSON etc. and since `java.sql.Types` does not have anything for that then just added new constants for those.